File: yntra-ios/YntraIOS/Views/DashboardView.swift
import SwiftUI
import yntra_core

struct DashboardView: View {
    @ObservedObject var viewModel: DashboardViewModel
    
    private let darkBackground = Color(red: 0.04, green: 0.06, blue: 0.1)
    private let cardBackground = Color(red: 0.12, green: 0.16, blue: 0.23)

    private var activeModules: [String: Bool] {
        guard let jsonStr = viewModel.workspace?.modulesActive,
              let data = jsonStr.data(using: .utf8),
              let dict = try? JSONSerialization.jsonObject(with: data, options: []) as? [String: Any] else {
            return [:]
        }
        return dict.compactMapValues { $0 as? Bool }
    }
    
    private var isTodosActive: Bool {
        viewModel.workspace == nil || activeModules["todos"] == true
    }
    
    private var isSchedulingActive: Bool {
        viewModel.workspace == nil || activeModules["scheduling"] == true
    }

    var body: some View {
        ZStack {
            darkBackground.ignoresSafeArea()
            
            // Glowing Ambient Bubbles
            VStack {
                HStack {
                    Circle()
                        .fill(Color(red: 0.31, green: 0.27, blue: 0.9).opacity(0.12))
                        .frame(width: 250, height: 250)
                        .blur(radius: 50)
                        .offset(x: -80, y: -50)
                    Spacer()
                }
                Spacer()
            }
            .ignoresSafeArea()

            ScrollView {
                VStack(alignment: .leading, spacing: 20) {
                    // Header
                    HStack {
                        VStack(alignment: .leading, spacing: 4) {
                            Text("Workspace Home")
                                .font(.system(size: 28, weight: .black, design: .rounded))
                                .foregroundColor(.white)
                            Text("Welcome back, Operator")
                                .font(.system(size: 13, weight: .medium, design: .rounded))
                                .foregroundColor(.white.opacity(0.6))
                        }
                        Spacer()
                    }
                    .padding(.horizontal)
                    .padding(.top, 10)
                    
                    if isTodosActive || isSchedulingActive {
                        // Stats grid
                        VStack(alignment: .leading, spacing: 16) {
                            Text("Active Modules Overview")
                                .font(.system(size: 16, weight: .bold, design: .rounded))
                                .foregroundColor(.white)
                            
                            HStack(spacing: 16) {
                                if isTodosActive {
                                    // Tasks widget
                                    VStack(alignment: .leading, spacing: 6) {
                                        Text("Tasks Done")
                                            .font(.caption)
                                            .foregroundColor(.white.opacity(0.5))
                                        Text("\(viewModel.completedTodosCount)/\(viewModel.todosCount)")
                                            .font(.system(size: 22, weight: .black, design: .rounded))
                                            .foregroundColor(.white)
                                    }
                                    .frame(maxWidth: .infinity, alignment: .leading)
                                    .padding()
                                    .background(cardBackground)
                                    .cornerRadius(16)
                                    .overlay(
                                        RoundedRectangle(cornerRadius: 16)
                                            .stroke(Color.white.opacity(0.05), lineWidth: 1)
                                    )
                                }
                                
                                if isSchedulingActive {
                                    // Events widget
                                    VStack(alignment: .leading, spacing: 6) {
                                        Text("Active Events")
                                            .font(.caption)
                                            .foregroundColor(.white.opacity(0.5))
                                        Text("\(viewModel.events.count)")
                                            .font(.system(size: 22, weight: .black, design: .rounded))
                                            .foregroundColor(.white)
                                    }
                                    .frame(maxWidth: .infinity, alignment: .leading)
                                    .padding()
                                    .background(cardBackground)
                                    .cornerRadius(16)
                                    .overlay(
                                        RoundedRectangle(cornerRadius: 16)
                                            .stroke(Color.white.opacity(0.05), lineWidth: 1)
                                    )
                                }
                            }
                        }
                        .padding(.horizontal)
                    }

                    if let errorMessage = viewModel.errorMessage {
                        Text("Error: \(errorMessage)")
                            .font(.caption)
                            .foregroundColor(.red)
                            .padding(.horizontal)
                    }

                    if isSchedulingActive {
                        // Schedule Events List
                        VStack(alignment: .leading, spacing: 12) {
                            Text("Upcoming Schedule / Timetable")
                                .font(.system(size: 16, weight: .bold, design: .rounded))
                                .foregroundColor(.white)
                            
                            if viewModel.events.isEmpty {
                                HStack {
                                    Spacer()
                                    VStack(spacing: 8) {
                                        Image(systemName: "calendar.badge.exclamationmark")
                                            .font(.system(size: 32))
                                            .foregroundColor(.white.opacity(0.3))
                                        Text("No upcoming events scheduled")
                                            .font(.system(size: 14, weight: .medium, design: .rounded))
                                            .foregroundColor(.white.opacity(0.5))
                                    }
                                    .padding(.vertical, 40)
                                    Spacer()
                                }
                                .background(cardBackground)
                                .cornerRadius(20)
                                .overlay(
                                    RoundedRectangle(cornerRadius: 20)
                                        .stroke(Color.white.opacity(0.05), lineWidth: 1)
                                )
                            } else {
                                ForEach(viewModel.events, id: \.id) { event in
                                    HStack {
                                        VStack(alignment: .leading, spacing: 4) {
                                            Text(event.title)
                                                .font(.system(size: 15, weight: .bold, design: .rounded))
                                                .foregroundColor(.white)
                                            Text("\(event.startTime) - \(event.endTime)")
                                                .font(.system(size: 12, weight: .medium, design: .rounded))
                                                .foregroundColor(.white.opacity(0.5))
                                        }
                                        Spacer()
                                        Image(systemName: "info.circle")
                                            .foregroundColor(Color.blue)
                                    }
                                    .padding(18)
                                    .background(cardBackground)
                                    .cornerRadius(18)
                                    .overlay(
                                        RoundedRectangle(cornerRadius: 18)
                                            .stroke(Color.white.opacity(0.05), lineWidth: 1)
                                    )
                                }
                            }
                        }
                        .padding(.horizontal)
                    }
                }
                .padding(.bottom, 20)
            }
        }
    }
}

File: yntra-android/app/src/main/java/com/yntra/app/views/DashboardView.kt
package com.yntra.app.views

import androidx.compose.foundation.background
import androidx.compose.foundation.layout.*
import androidx.compose.foundation.lazy.LazyColumn
import androidx.compose.foundation.lazy.items
import androidx.compose.foundation.shape.RoundedCornerShape
import androidx.compose.material.icons.Icons
import androidx.compose.material.icons.filled.DateRange
import androidx.compose.material.icons.filled.Info
import androidx.compose.material3.*
import androidx.compose.runtime.*
import androidx.compose.ui.Alignment
import androidx.compose.ui.Modifier
import androidx.compose.ui.graphics.Brush
import androidx.compose.ui.graphics.Color
import androidx.compose.ui.text.font.FontWeight
import androidx.compose.ui.unit.dp
import androidx.compose.ui.unit.sp
import com.yntra.app.viewmodels.DashboardViewModel

@OptIn(ExperimentalMaterial3Api::class)
@Composable
fun DashboardView(viewModel: DashboardViewModel) {
    val workspace by viewModel.workspace.collectAsState()
    val events by viewModel.events.collectAsState()
    val todosCount by viewModel.todosCount.collectAsState()
    val completedTodosCount by viewModel.completedTodosCount.collectAsState()
    val errorMessage by viewModel.errorMessage.collectAsState()

    val activeModules = remember(workspace) {
        val modules = mutableMapOf<String, Boolean>()
        workspace?.modules_active?.let { jsonStr ->
            try {
                val jsonObj = org.json.JSONObject(jsonStr)
                val keys = jsonObj.keys()
                while (keys.hasNext()) {
                    val key = keys.next()
                    modules[key] = jsonObj.optBoolean(key, false)
                }
            } catch (e: Exception) {
                // Ignore parsing errors
            }
        }
        modules
    }

    val isTodosActive = workspace == null || activeModules["todos"] == true
    val isSchedulingActive = workspace == null || activeModules["scheduling"] == true

    Column(
        modifier = Modifier
            .fillMaxSize()
            .background(Color(0xFF0B0F19))
            .padding(20.dp),
        verticalArrangement = Arrangement.spacedBy(16.dp)
    ) {
        // Welcome Header
        Row(
            modifier = Modifier.fillMaxWidth(),
            horizontalArrangement = Arrangement.SpaceBetween,
            verticalAlignment = Alignment.CenterVertically
        ) {
            Column {
                Text(
                    text = "Workspace Home",
                    color = Color.White,
                    fontSize = 26.sp,
                    fontWeight = FontWeight.Black
                )
                Text(
                    text = "Welcome back, Operator",
                    color = Color(0xFF94A3B8),
                    fontSize = 13.sp
                )
            }
        }

        // Stats summary block
        if (isTodosActive || isSchedulingActive) {
            Card(
                colors = CardDefaults.cardColors(containerColor = Color(0xFF1E293B)),
                shape = RoundedCornerShape(20.dp),
                modifier = Modifier.fillMaxWidth()
            ) {
                Column(
                    modifier = Modifier.padding(20.dp),
                    verticalArrangement = Arrangement.spacedBy(12.dp)
                ) {
                    Text(
                        text = "Active Modules Overview",
                        color = Color.White,
                        fontWeight = FontWeight.Bold,
                        fontSize = 16.sp
                    )
                    
                    Row(
                        modifier = Modifier.fillMaxWidth(),
                        horizontalArrangement = Arrangement.spacedBy(16.dp)
                    ) {
                        // Todos Card
                        if (isTodosActive) {
                            Card(
                                colors = CardDefaults.cardColors(containerColor = Color(0xFF334155)),
                                modifier = Modifier.weight(1f),
                                shape = RoundedCornerShape(12.dp)
                            ) {
                                Column(
                                    modifier = Modifier.padding(14.dp)
                                ) {
                                    Text(text = "Tasks Done", color = Color(0xFF94A3B8), fontSize = 11.sp)
                                    Spacer(modifier = Modifier.height(4.dp))
                                    Text(text = "$completedTodosCount/$todosCount", color = Color.White, fontSize = 20.sp, fontWeight = FontWeight.Bold)
                                }
                            }
                        }

                        // Events Card
                        if (isSchedulingActive) {
                            Card(
                                colors = CardDefaults.cardColors(containerColor = Color(0xFF334155)),
                                modifier = Modifier.weight(1f),
                                shape = RoundedCornerShape(12.dp)
                            ) {
                                Column(
                                    modifier = Modifier.padding(14.dp)
                                ) {
                                    Text(text = "Active Events", color = Color(0xFF94A3B8), fontSize = 11.sp)
                                    Spacer(modifier = Modifier.height(4.dp))
                                    Text(text = "${events.size}", color = Color.White, fontSize = 20.sp, fontWeight = FontWeight.Bold)
                                }
                            }
                        }
                    }
                }
            }
        }

        // Error message card
        errorMessage?.let {
            Card(
                colors = CardDefaults.cardColors(containerColor = MaterialTheme.colorScheme.errorContainer),
                shape = RoundedCornerShape(12.dp)
            ) {
                Text(
                    text = it,
                    color = MaterialTheme.colorScheme.onErrorContainer,
                    modifier = Modifier.padding(12.dp),
                    fontSize = 14.sp
                )
            }
        }

        // Events/Timetable section
        if (isSchedulingActive) {
            Text(
                text = "Upcoming Schedule / Timetable",
                color = Color.White,
                fontWeight = FontWeight.Bold,
                fontSize = 16.sp,
                modifier = Modifier.padding(top = 8.dp)
            )

            if (events.isEmpty()) {
                Box(
                    modifier = Modifier
                        .fillMaxWidth()
                        .weight(1f)
                        .background(Color(0xFF1E293B), RoundedCornerShape(16.dp)),
                    contentAlignment = Alignment.Center
                ) {
                    Column(
                        horizontalAlignment = Alignment.CenterHorizontally,
                        verticalArrangement = Arrangement.spacedBy(8.dp)
                    ) {
                        Icon(
                            imageVector = Icons.Default.DateRange,
                            contentDescription = "Empty",
                            tint = Color(0xFF94A3B8),
                            modifier = Modifier.size(36.dp)
                        )
                        Text(text = "No upcoming events scheduled", color = Color(0xFF94A3B8), fontSize = 14.sp)
                    }
                }
            } else {
                LazyColumn(
                    modifier = Modifier.weight(1f),
                    verticalArrangement = Arrangement.spacedBy(10.dp)
                ) {
                    items(events) { event ->
                        Card(
                            colors = CardDefaults.cardColors(containerColor = Color(0xFF1E293B)),
                            shape = RoundedCornerShape(16.dp)
                        ) {
                            Row(
                                modifier = Modifier
                                    .fillMaxWidth()
                                    .padding(16.dp),
                                verticalAlignment = Alignment.CenterVertically
                            ) {
                                Column(
                                    modifier = Modifier.weight(1f),
                                    verticalArrangement = Arrangement.spacedBy(4.dp)
                                ) {
                                    Text(
                                        text = event.title,
                                        color = Color.White,
                                        fontSize = 15.sp,
                                        fontWeight = FontWeight.SemiBold
                                    )
                                    Text(
                                        text = "${event.startTime} - ${event.endTime}",
                                        color = Color(0xFF94A3B8),
                                        fontSize = 12.sp
                                    )
                                }
                                
                                Icon(
                                    imageVector = Icons.Default.Info,
                                    contentDescription = "Detail",
                                    tint = Color(0xFF8B5CF6)
                                )
                            }
                        }
                    }
                }
            }
        }
    }
}
